tb_pulumi/cloudwatch.py

```diff
@@ -0,0 +1,281 @@
+            resources={'sns_topic': sns_topic, 'sns_subscriptions': subscriptions, 'alarms': alarms},
+        )
+
+
+class AlbAlarmGroup(tb_pulumi.monitoring.AlarmGroup):
+    """A set of alarms for Application Load Balancers. Contains the following configurable alarms:
+
+        - ``fivexx``: Alarms on the number of HTTP responses with status codes in the 500-599 range, indicating a count
+        of internal server errors.
+        - ``response_time`: Alarms on the response time of HTTP requests. Threshold should be set in seconds.
+
+    :param name: The name of the the ``AlbAlarmGroup`` resource.
+    :type name: str
+
+    :param monitoring_group: The ``CloudWatchMonitoringGroup`` this alarm group belongs to.
+    :type monitoring_group: CloudWatchMonitoringGroup
+
+    :param project: The ``ThunderbirdPulumiProject`` being monitored.
+    :type project: tb_pulumi.ThunderbirdPulumiProject
+
+    :param resource: The ``aws.lb.load_balancer.LoadBalancer`` being monitored.
+    :type resource: aws.lb.load_balancer.LoadBalancer
+
+    :param opts: Additional ``pulumi.ResourceOptions`` to apply to this resource. Defaults to None.
+    :type opts: pulumi.ResourceOptions, optional
+    """
+
+    def __init__(
+        self,
+        name: str,
+        monitoring_group: CloudWatchMonitoringGroup,
+        project: tb_pulumi.ThunderbirdPulumiProject,
+        resource: aws.lb.load_balancer.LoadBalancer,
+        opts: pulumi.ResourceOptions = None,
+        **kwargs,
+    ):
+        super().__init__(
+            pulumi_type='tb:cloudwatch:AlbAlarmGroup',
+            name=name,
+            monitoring_group=monitoring_group,
+            project=project,
+            resource=resource,
+            opts=opts,
+            **kwargs,
+        )
+
+        # Alert if we see sustained 5xx statuses
+        fivexx_opts = {'enabled': True, 'evaluation_periods': 2, 'period': 300, 'statistic': 'Average', 'threshold': 10}
+        fivexx_opts.update(self.overrides['fivexx'] if 'fivexx' in self.overrides else {})
+        fivexx_enabled = fivexx_opts['enabled']
+        del fivexx_opts['enabled']
+        fivexx = pulumi.Output.all(res_name=resource.name, res_suffix=resource.arn_suffix).apply(
+            lambda outputs: aws.cloudwatch.MetricAlarm(
+                f'{self.name}-5xx',
+                name=f'{self.project.name_prefix}-5xx',
+                comparison_operator='GreaterThanOrEqualToThreshold',
+                dimensions={'LoadBalancer': f'app/{outputs['res_name']}/{outputs['res_suffix']}'},
+                metric_name='HTTPCode_ELB_5XX_Count',
+                namespace='AWS/ApplicationELB',
+                alarm_description=f'Elevated 5xx errors on ALB {outputs['res_name']}',
+                opts=pulumi.ResourceOptions(parent=self, depends_on=[resource]),
+                **fivexx_opts,
+            )
+            if fivexx_enabled
+            else None
+        )
+        fivexx = ()
```

Comment:
The above pattern allows for reasonable defaults with optional overrides. This ends up being a little ugly in this module's code, but the user experience in the end is one involving much less effort.